src/gstreamer/gstkvssink.cpp

```diff
@@ -1033,12 +1055,30 @@ void create_kinesis_video_frame(Frame *frame, const nanoseconds &pts, const nano
     frame->trackId = static_cast<UINT64>(track_id);
 }
 
-bool
-put_frame(shared_ptr<KinesisVideoStream> kinesis_video_stream, void *frame_data, size_t len, const nanoseconds &pts,
+bool put_frame(GstKvsSink *kvsSink, void *frame_data, size_t len, const nanoseconds &pts,
           const nanoseconds &dts, FRAME_FLAGS flags, uint64_t track_id, uint32_t index) {
+    if(kvsSink == nullptr){
+        GST_ERROR_OBJECT (kvsSink, "Missing User Data");
+        LOG_INFO("Missing User Data");
+        return FALSE;
+    }
     Frame frame;
     create_kinesis_video_frame(&frame, pts, dts, flags, frame_data, len, track_id, index);
-    return kinesis_video_stream->putFrame(frame);
+    bool ret = kvsSink->data->kinesis_video_stream->putFrame(frame);
+    if(CHECK_FRAME_FLAG_KEY_FRAME(flags)  || (kvsSink->data->onFirstFrame && ret)){
+        KvsSinkMetric *kvsSinkMetric = new KvsSinkMetric();
+        kvsSinkMetric->streamMetrics = kvsSink->data->kinesis_video_stream->getMetrics();
+        kvsSinkMetric->clientMetrics = kvsSink->data->kinesis_video_producer->getMetrics();
+        kvsSinkMetric->framePTS = frame.presentationTs;
+        kvsSinkMetric->onFirstFrame = kvsSink->data->onFirstFrame;
+        kvsSinkMetric->putFrameSuccess = ret;
```

Comment:
Do we track putFrameSuccess or would customers need this? Given we are invoking this signal only on success, we shouldnt need this.